examples/BoundaryConditionsCases/configurable_constraint_allow_yaw.py

```diff
@@ -0,0 +1,140 @@
+__doc__ = """Fixed joint example, for detailed explanation refer to Zhang et. al. Nature Comm.  methods section."""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import sys
+
+# FIXME without appending sys.path make it more generic
+sys.path.append("../../")
+from elastica import *
+from examples.BoundaryConditionsCases.bc_cases_postprocessing import (
+    plot_position,
+    plot_orientation,
+    plot_video,
+    plot_video_xy,
+    plot_video_xz,
+)
+
+
+class ConfigurableConstraintSimulator(
+    BaseSystemCollection, Constraints, Connections, Forcing, Damping, CallBacks
+):
+    pass
+
+
+configurable_constraint_sim = ConfigurableConstraintSimulator()
+
+# setting up test params
+n_elem = 10
+direction = np.array([0.0, 0.0, 1.0])
+normal = np.array([0.0, 1.0, 0.0])
+base_length = 0.2
+base_radius = 0.007
+base_area = np.pi * base_radius ** 2
+density = 1750
+E = 3e7
+poisson_ratio = 0.5
+shear_modulus = E / (poisson_ratio + 1.0)
+
+# setting up timestepper and video
+final_time = 10
+dl = base_length / n_elem
+dt = 1e-5
+total_steps = int(final_time / dt)
+fps = 100  # frames per second of the video
+diagnostic_step_skip = 1 / (fps * dt)
+
+start_rod_1 = np.zeros((3,))
+start_rod_2 = start_rod_1 + direction * base_length
+
+# Create rod 1
+rod1 = CosseratRod.straight_rod(
+    n_elem,
+    start_rod_1,
+    direction,
+    normal,
+    base_length,
+    base_radius,
+    density,
+    0.0,  # internal damping constant, deprecated in v0.3.0
+    E,
+    shear_modulus=shear_modulus,
+)
+configurable_constraint_sim.append(rod1)
+
+# Apply boundary conditions to rod1: only allow yaw
+configurable_constraint_sim.constrain(rod1).using(
+    ConfigurableConstraint,
```

Comment:
```suggestion
    GeneralConstraint,
```

Reply:
Implemented in https://github.com/GazzolaLab/PyElastica/pull/143/commits/4c06b0e850b5d5fb647394771572c35a9f0c641e